### PR TITLE
Use informative error message for optional dependency imports

### PR DESCRIPTION
**Pull request by maxrjones**
**Description of proposed changes**

This PR adds information about optional dependencies to the installation instructions and improves the ImportError message when trying to use the data loaders without having PyTorch and/or TensorFlow installed.

Open to opinions about whether these error messages should also be used for the torch and tf xarray accessors and if it's necessary to add tests for this case. This drops code coverage to 99%.

Fixes #

### CODE REVIEW

File: xbatcher/loaders/keras.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 from typing import Any, Callable, Optional, Tuple
 
-import tensorflow as tf
+try:
+    import tensorflow as tf
+except ImportError:
+    raise ImportError(
+        "The Xbatcher TensorFlow Dataset API depends on TensorFlow. Please "
+        "install TensorFlow to proceed."
+    )
```
Comment (andersy005) [inline xbatcher/loaders/keras.py:9]: Raise from original exception??

```suggestion
except ImportError as exc:
    raise ImportError(
        "The Xbatcher TensorFlow Dataset API depends on TensorFlow. Please "
        "install TensorFlow to proceed."
    ) from exc
```

File: xbatcher/loaders/torch.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
 from typing import Any, Callable, Optional, Tuple
 
-import torch
+try:
+    import torch
+except ImportError:
+    raise ImportError(
+        "The Xbatcher PyTorch Dataset API depends on PyTorch. Please "
+        "install PyTorch to proceed."
+    )
```
Comment (andersy005) [inline xbatcher/loaders/torch.py:9]:
```suggestion
except ImportError as exc:
    raise ImportError(
        "The Xbatcher PyTorch Dataset API depends on PyTorch. Please "
        "install PyTorch to proceed."
    ) from exc
```

File: xbatcher/loaders/keras.py (hunk 1 above)
Comment (maxrjones) [reply xbatcher/loaders/keras.py:9]: Good catch, thanks!

**Review by andersy005 — APPROVED** [review]
Looks good to me 👍🏽. Thank you, @maxrjones

**Pull request merged**